doc/i2p.md

```diff
@@ -47,9 +52,35 @@ In a typical situation, this suffices:
 bitcoind -i2psam=127.0.0.1:7656
 ```
 
-The first time Bitcoin Core connects to the I2P router, its I2P address (and
-corresponding private key) will be automatically generated and saved in a file
-named `i2p_private_key` in the Bitcoin Core data directory.
+The first time Bitcoin Core connects to the I2P router, if a persistent I2P
+address is to be used, then it (and its corresponding private key) will be
+automatically generated and saved in a file named `i2p_private_key` in the
+Bitcoin Core data directory. A persistent I2P address is used for accepting
+incoming connections (`-i2pacceptincoming`) and for making outgoing connections
+if `-i2ptransientout` is not used.
+
+## Persistent vs transient I2P addresses
+
+In I2P connections, the connection receiver sees the I2P address of the
+connection initiator. This is unlike the Tor network where the recipient does
+not know who is connecting to them, not even the initiator's Tor address.
+
+Even if an I2P node is not accepting incoming connections, it is known to other
+nodes by its outgoing I2P address. If this address is always the same, then
+there is an opportunity to white-list a give node. However, this also creates an
+opportunity to discriminate, fingerprint or analyze a given node based on its
+I2P address.
+
+For this reason, the `-i2ptransientout` configuration option generates and
+uses a new, transient (disposable) I2P address for each new outbound connection.
+The address will never be reused, not even if reconnecting to the same peer
+later.
+
+`-i2ptransientout` only affects addresses for outgoing connections and not the
+address for accepting incoming connections (if `-i2pacceptincoming` is given).
+This is because the address for accepting connections is advertised to the
+Bitcoin network and it takes time for it to propagate and be known by other
+nodes. So, it does not make sense to change it often - it must be persistent.
```

Comment:
Don't begin a sentence with "This is because..." (suggest dropping it: "The address...")

Idem for "So, ..." (suggest dropping)

An [emdash](https://www.thepunctuationguide.com/em-dash.html) (" — " or " --- ") should be used here, not a hyphen. An emdash can optionally be set off with spaces.

```suggestion
nodes. It does not make sense to change it often — it must be persistent.
```

Reply:
Done, changed the minus to em-_and_ :laughing: